### ISSUE TITLE
[REQUEST] EmberZNet 7.4.x.x Zigbee Router device firmware image for Sonoff Zigbee 3.0 USB Dongle Plus V2 (model “ZBDongle-E”) by ITead?

### ISSUE THREAD

**Issue by Hedda** [itead/Sonoff_Zigbee_Dongle_Firmware#33]
@Daniel-zhan-itead @liangjia2019 @guozi7788 I like to suggest that ITead also releasing a new official Zigbee Router device firmware mage based on latest EmberZNet NCP 7.4.x.0 version for your EFR32MG21 based  [“Sonoff Zigbee 3.0 USB Dongle Plus V2”](https://itead.cc/product/zigbee-3-0-usb-dongle/) (model [“ZBDongle-E”, a.k.a. "Dongle-E”](https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-e/)) as an optional firmware:

https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/Router

Note that the current Zigbee Router firmware based on EmberZNet 6.10.3.0 is working great and is already much appriciacted, it is just that it would be nice if the was an optional updated Zigbee Router firmware version that can be tested (as experimental at first before it can recommended).

EmberZNet 6.10.3.0 version is going on in age and even trough it works great a newer version should hopefully provide bug-fixes, etc..

https://github.com/SiliconLabs/gecko_sdk/releases

**Comment by jorisvervuurt** [comment]
Same, I would greatly appreciate a new router firmware build based on 7.4. The current router firmware is over 2 years old. @liangjia2019

**Comment by liangjia2019** [comment]
> Same, I would greatly appreciate a new router firmware build based on 7.4. The current router firmware is over 2 years old. @liangjia2019

It has been put on the agenda, and I will push the team to push the process. You will see his update in the near future

**Comment by Hedda** [comment]
FYI, puddly, darkxst, and Nerivec have together been working on adding avility for build Zigbee Router firmware with NabuCasa’s silabs-firmware-builder:

https://github.com/NabuCasa/silabs-firmware-builder/pull/85

Nerivec Is still experimenting with this so you can test his unofficial builds:

https://github.com/Nerivec/silabs-firmware-builder/discussions/16

For reference, they intent to enable use ZAP tooling/config with custom [silabs-firmware-builder](https://github.com/NabuCasa/silabs-firmware-builder) to build Zigbee Router and other custom firmware for different Silicon Labs based radio adapters:

* https://github.com/NabuCasa/silabs-firmware-builder/pull/84

> _Implement a new protocol for communicating with the builtin CLI on Silabs router firmware and integrate through out. This will allow to probe and flash router firmware_
>
> _Router firmware must implement a custom CLI command bootloader reboot that calls the bootloader_rebootAndInstall api._
>
> _This has been tested both standalone and integrated with sl-web-tools._

>  _Probe of CPC 115200 is fixed in https://github.com/NabuCasa/universal-silabs-flasher/pull/86, but also requires router firmware built with SL_CLI_LOCAL_ECHO: 0_


ZAP in this case refers to the ZCL Advanced Platform (ZAP) tool/format for configuring Matter devices in a standardized way:

* https://github.com/project-chip/zap

* https://docs.silabs.com/matter/2.2.2/matter-references/matter-zap

* https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/protocols/matter/getting_started/tools.html

**Comment by Hedda** [comment]
> FYI, puddly, darkxst, and Nerivec have together been working on adding the ability for build Zigbee Router firmware with NabuCasa’s silabs-firmware-builder:
> 
> [NabuCasa/silabs-firmware-builder#85](https://github.com/NabuCasa/silabs-firmware-builder/pull/85)

@liangjia2019 a good way to prepare would be to start using NabuCasa’s silabs-firmware-builder tool to build other firmware images to get used to that process and allow building firmware images more regularly. See more details about that in this other issue discussion:

- https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/issues/17

**Comment by liangjia2019** [comment]
> Same, I would greatly appreciate a new router firmware build based on 7.4. The current router firmware is over 2 years old. @liangjia2019

@jorisvervuurt  
Latest update: Our team is currently testing the latest 7.4.x version of the Router firmware. If you're interested, we can send it to you for early experience via email.

**Comment by Hedda** [comment]
@liangjia2019 as mentioned above, before you put too much effort into test router firmware based on EmberZNet 7.4.x you should note that Nerivec have published (unofficial) EmberZet 8.0.0.x Zigbee Router for Sonoff and other USB dongles: 

* https://github.com/Nerivec/silabs-firmware-builder/discussions/16

While 7.4 has proven stable as Zigbee Coordinator firmware it might now be a good idea if you skip 7.x and instead target EmberZNet 8.x version for the Zigbee Router firmware.

PS:  For reference as might relevant to this; there was a known bug with UART buffering (RX buffer) timeout issues when using software flow control with all the current versions of EmberZNet 7.4.x, 8.0.x, and 8.1.x but not sure if this is suppose to fix that? That bug does affect all of the existing EFR32MG21-based Zigbee Coordinator adapter that has a [CP2102N/CP2102 (CP210x) USB to UART Bridge VCP](https://www.silabs.com/developer-tools/usb-to-uart-bridge-vcp-drivers), and the only know workaround for that bug has been to build a new firmware image that increase default SL_IOSTREAM_USART_VCOM_RX_BUFFER_SIZE from 32 to 128 (or above).

* https://github.com/SiliconLabs/simplicity_sdk/issues/29

PPS: FYI, Nerivec does provide unofficial experimental firmware builds on GitHub in his fork of Nabu Casa's silabs-firmware-builder which has implemented that VCOM RX bufffer size increase workaround:

* https://github.com/Nerivec/silabs-firmware-builder
  * https://github.com/Nerivec/silabs-firmware-builder/releases
    * https://github.com/Nerivec/silabs-firmware-builder/discussions

**Comment by jorisvervuurt** [comment]
Yes, I have now moved to SMLight SLZB-07MG24 dongles with Nerivec's latest firmware builds (both coordinator and router). Works great!

**Comment by liangjia2019** [comment]
> @liangjia2019 as mentioned above, before you put too much effort into test router firmware based on EmberZNet 7.4.x you should note that Nerivec have published (unofficial) EmberZet 8.0.0.x Zigbee Router for Sonoff and other USB dongles:
> 
> * [Zigbee Router: Feedback wanted! Nerivec/silabs-firmware-builder#16](https://github.com/Nerivec/silabs-firmware-builder/discussions/16)
> 
> While 7.4 has proven stable as Zigbee Coordinator firmware it might now be a good idea if you skip 7.x and instead target EmberZNet 8.x version for the Zigbee Router firmware.

Thank you for your suggestion. We have also evaluated internally whether to skip the 7.x version and directly release the 8.x version of the firmware. The final decision is to provide both versions, giving users the option to choose.
